crates/l2/contracts/src/l2/L2Upgradeable.sol

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.29;
+
+import "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
+import "@openzeppelin/contracts/proxy/ERC1967/ERC1967Utils.sol";
+
+/// @title Interface for an L2-capable token.
+/// @author LambdaClass
+/// @dev Uses the interface described in the ERC-7802 draft
+contract UpgradeableSystemContract is ERC1967Proxy {
```

Comment:
Can we use a common `TransparentUpgradeableProxy` instead of rolling our own implementation?

Reply:
Do we want Transparent Proxy or UUPS? L1 contracts use UUPS for more context